BuilderPatternExample03/src/com/gmail/granovskiy/s/BankAccount.java

```diff
@@ -0,0 +1,113 @@
+package com.gmail.granovskiy.s;
+
+public class BankAccount {
+    private long accountNumber;
+    private String owner;
+    private String branch;
+    private double balance;
+    private double interestRate;
+
+    private BankAccount(long accountNumber, String owner, String branch, double balance, double interestRate) {
+        this.accountNumber = accountNumber;
+        this.owner = owner;
+        this.branch = branch;
+        this.balance = balance;
+        this.interestRate = interestRate;
+    }
+
+    //Getters and setters
+    public long getAccountNumber() {
+        return accountNumber;
+    }
+
+    public void setAccountNumber(long accountNumber) {
+        this.accountNumber = accountNumber;
+    }
+
+    public String getOwner() {
+        return owner;
+    }
+
+    public void setOwner(String owner) {
+        this.owner = owner;
+    }
+
+    public String getBranch() {
+        return branch;
+    }
+
+    public void setBranch(String branch) {
+        this.branch = branch;
+    }
+
+    public double getBalance() {
+        return balance;
+    }
+
+    public void setBalance(double balance) {
+        this.balance = balance;
+    }
+
+    public double getInterestRate() {
+        return interestRate;
+    }
+
+    public void setInterestRate(double interestRate) {
+        this.interestRate = interestRate;
+    }
+
+    //  Class Builder
+    public static class Builder {
+        private long accountNumber;
+        private String owner;
+        private String branch;
+        private double balance;
+        private double interestRate;
+
+        public Builder(long accountNumber) {
+            this.accountNumber = accountNumber;
+        }
+
+        public Builder withOwner(String owner) {
+            this.owner = owner;
+            return this;
+        }
+
+        public Builder atBranch(String branch) {
+            this.branch = branch;
+            return this;
+        }
+
+        public Builder openingBalance(double balance) {
+            this.balance = balance;
+            return this;
+        }
+
+        public Builder atRate(double interestRate) {
+            this.interestRate = interestRate;
+            return this;
+        }
+
+        public BankAccount build() {
+            //Here we create the actual bank account object, which is always in a fully initialised state when it's returned.
+            BankAccount account = new BankAccount(3456789L, "Mike", "SouthBranch", 1000000.00, 50.00);
+            account.accountNumber = this.accountNumber;
+            account.owner = this.owner;
+            account.branch = this.branch;
```

Comment:
here should be just -->  return new BankAccount(accountNumber, owner, branch, balance, interestRate);